packages/experiments/src/components/Toggle/Toggle.styles.ts

```diff
@@ -208,15 +199,33 @@ export const ToggleStyles: IToggleComponent['styles'] = props => {
     return {};
   }
 
-  function getToggleStylesFromVariant(variantVariables: { [PState in IToggleStates]: IToggleStyleVariablesTypes }): Partial<IToggleStyles> {
-    return concatStyleSets(
-      getToggleStylesFromState(variantVariables.baseState),
-      !disabled && !checked && getToggleStylesFromState(variantVariables.enabled),
-      disabled && !checked && getToggleStylesFromState(variantVariables.disabled),
-      !disabled && checked && getToggleStylesFromState(variantVariables.checked),
-      disabled && checked && getToggleStylesFromState(variantVariables.checkedDisabled)
-    );
+  function getToggleViewStyleVariables(viewProps: IToggleViewProps): IToggleStyleVariablesTypes {
+    let toggleVariables: IToggleStyleVariablesTypes = toggleBaseStateVariables;
+
+    toggleVariables = {
+      ...toggleVariables,
+      ...(viewProps.checked ? toggleCheckedVariables : {}),
+      ...(viewProps.disabled
+        ? { ...toggleDisabledVariables, ...(viewProps.checked ? toggleCheckedDisabledVariables : {}) }
+        : { ...toggleEnabledVariables, ...(viewProps.checked ? toggleCheckedEnabledVariables : {}) })
+    };
+
+    return toggleVariables;
+  }
+
+  function getToggleStyles(userStyleVariables: IToggleStyleVariables): Partial<IToggleStyles> {
+    let toggleVariables: IToggleStyleVariablesTypes;
+
+    toggleVariables = getToggleViewStyleVariables(props);
+
+    if (typeof userStyleVariables === 'function') {
+      toggleVariables = { ...toggleVariables, ...userStyleVariables(props) };
+    } else if (userStyleVariables !== undefined) {
+      toggleVariables = { ...toggleVariables, ...userStyleVariables };
+    }
```

Comment:
This is pretty straightforward but this would be a good util function, assuming there isn't one already.